dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/registry/HeartBeatTask.java

```diff
@@ -67,6 +67,15 @@ public HeartBeatTask(long startupTime,
         this.heartBeat = new HeartBeat(startupTime, maxCpuloadAvg, reservedMemory, hostWeight, workerThreadCount);
     }
 
+    public void setWorkerWaitingTaskCount(int count){
+        this.workerWaitingTaskCount  = count;
+    }
+
+    public void incrementWorkerWaitingTaskCount(){
+        this.workerWaitingTaskCount++;
```

Comment:
I don't think this modification can solve this problem, this value is the number of tasks to be run held by the current worker, not the count of heartbeat

Reply:
I suggest splitting this class into master and worker to obtain it separately. BTW, please test it fully

Reply:
+1， this patch cannot solve this problem, you can use a Supplier to replace the workerWaitingTaskCount to solve this.